domain/src/test/kotlin/no/nav/su/se/bakover/domain/satser/SatsFactoryForSupplerendeStønadUføreTest.kt

```diff
@@ -267,81 +294,98 @@ internal class SatsFactoryForSupplerendeStønadTest {
 
-            val factoryMai19TilMai20 = verdier[0]
-            val factoryMai20TilMai21 = verdier[1]
+            val factoryMai19TilSeptember20 = verdier[0]
+            val factorySeptember20TilMai21 = verdier[1]
             val factoryMai21TilMai22 = verdier[2]
 
-            factoryMai19TilMai20.gjeldende(1.juni(2019)) shouldBeSameInstanceAs factoryMai19TilMai20
-            factoryMai19TilMai20.gjeldende(1.mars(2020)) shouldBeSameInstanceAs factoryMai19TilMai20
-            factoryMai20TilMai21.gjeldende(1.juni(2020)) shouldBeSameInstanceAs factoryMai20TilMai21
-            factoryMai20TilMai21.gjeldende(1.mars(2021)) shouldBeSameInstanceAs factoryMai20TilMai21
-            factoryMai21TilMai22.gjeldende(1.juni(2021)) shouldBeSameInstanceAs factoryMai21TilMai22
-            factoryMai21TilMai22.gjeldende(1.mars(2022)) shouldBeSameInstanceAs factoryMai21TilMai22
+            satsFactoryTest.gjeldende(1.mai(2019)) shouldBeSameInstanceAs factoryMai19TilSeptember20
+            satsFactoryTest.gjeldende(3.september(2020)) shouldBeSameInstanceAs factoryMai19TilSeptember20
+            satsFactoryTest.gjeldende(4.september(2020)) shouldBeSameInstanceAs factorySeptember20TilMai21
+            satsFactoryTest.gjeldende(20.mai(2021)) shouldBeSameInstanceAs factorySeptember20TilMai21
+            satsFactoryTest.gjeldende(21.mai(2021)) shouldBeSameInstanceAs factoryMai21TilMai22
+            satsFactoryTest.gjeldende(19.mai(2022)) shouldBeSameInstanceAs factoryMai21TilMai22
+            satsFactoryTest.gjeldende(19.mai(2022)) shouldBeSameInstanceAs factoryMai21TilMai22
 
-            factoryMai19TilMai20.høy(januar(2022)).grunnbeløp shouldBe GrunnbeløpForMåned(
+            factoryMai19TilSeptember20.høyUføre(januar(2022)).grunnbeløp shouldBe GrunnbeløpForMåned(
                 måned = januar(2022),
                 grunnbeløpPerÅr = 99858,
                 ikrafttredelse = 1.mai(2019),
                 virkningstidspunkt = 1.mai(2019),
             )
-            factoryMai20TilMai21.høy(januar(2022)).grunnbeløp shouldBe GrunnbeløpForMåned(
+            factorySeptember20TilMai21.høyUføre(januar(2022)).grunnbeløp shouldBe GrunnbeløpForMåned(
                 måned = januar(2022),
                 grunnbeløpPerÅr = 101351,
-                ikrafttredelse = 1.mai(2020),
+                ikrafttredelse = 4.september(2020),
                 virkningstidspunkt = 1.mai(2020),
             )
-            factoryMai21TilMai22.høy(januar(2022)).grunnbeløp shouldBe GrunnbeløpForMåned(
+            factoryMai21TilMai22.høyUføre(januar(2022)).grunnbeløp shouldBe GrunnbeløpForMåned(
                 måned = januar(2022),
                 grunnbeløpPerÅr = 106399,
-                ikrafttredelse = 1.mai(2021),
+                ikrafttredelse = 21.mai(2021),
                 virkningstidspunkt = 1.mai(2021),
             )
         }
+
+        @Test
+        fun `Cache tar høyde for knekkpunkter på tvers av satser og grunnbeløps ikraftredelser`() {
```

Comment:
Sjekkes det noe "på tvers" her? Burde kanskje sjekket at `satsFactory.gjeldende(1.januar(2016)` returnerer noe annet for andre ting? Sikkert enklest hvis man konstruerer kunstige cases.

Reply:
Utvidet testen litt. Fant ut at vi støtter dårlig duplikate/usorterte satsdata, så la inn litt sperrer på dette :)